CMakeLists.txt

```diff
@@ -159,75 +165,117 @@ set(XWIDGETS_SOURCES
     ${XWIDGETS_SOURCE_DIR}/xvideo.cpp
 )
 
-# Output
-# ======
+# Targets and link
+# ================
+
+include(CheckCXXCompilerFlag)
+
+macro(xwidgets_create_target target_name linkage output_name)
+    string(TOUPPER "${linkage}" linkage_upper)
 
-add_library(xwidgets SHARED ${XWIDGETS_SOURCES} ${XWIDGETS_HEADERS})
-target_include_directories(xwidgets PUBLIC $<BUILD_INTERFACE:${XWIDGETS_INCLUDE_DIR}>
-                                           $<INSTALL_INTERFACE:include>)
+    if (NOT ${linkage_upper} MATCHES "^(SHARED|STATIC)$")
+        message(FATAL_ERROR "Invalid library linkage: ${linkage}")
+    endif ()
 
-target_link_libraries(xwidgets
-                      PUBLIC xtl
-                      PUBLIC xeus)
+    if (${linkage_upper} STREQUAL "STATIC")
+        set(XEUS_TARGET_NAME xeus-static)
+    else()
+        set(XEUS_TARGET_NAME xeus)
+    endif()
 
-set_target_properties(xwidgets PROPERTIES
-                      PUBLIC_HEADER "${XWIDGETS_HEADERS}"
-                      COMPILE_DEFINITIONS "XWIDGETS_EXPORTS"
-                      PREFIX ""
-                      VERSION ${XWIDGETS_BINARY_VERSION}
-                      SOVERSION ${XWIDGETS_BINARY_CURRENT}
-                      OUTPUT_NAME "libxwidgets")
+    # Output
+    # ======
 
-# Compilation flags
-# =================
+    add_library(${target_name} ${linkage_upper} ${XWIDGETS_SOURCES} ${XWIDGETS_HEADERS})
+    target_include_directories(${target_name} PUBLIC $<BUILD_INTERFACE:${XWIDGETS_INCLUDE_DIR}>
+                                               $<INSTALL_INTERFACE:include>)
 
-include(CheckCXXCompilerFlag)
-string(TOUPPER "${CMAKE_BUILD_TYPE}" U_CMAKE_BUILD_TYPE)
-OPTION(DISABLE_ARCH_NATIVE "disable -march=native flag" OFF)
+    target_link_libraries(${target_name}
+                          PUBLIC xtl
+                          PUBLIC ${XEUS_TARGET_NAME}
+                          PUBLIC xproperty)
```

Comment:
This was necessary to allow builds with in-tree xproperty (obtained with cmake's FetchContent). Without it all includes to xproperty files are not found

Reply:
:+1: